packages/@react-aria/focus/src/useHasTabbableChild.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2022 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {getFocusableTreeWalker} from './FocusScope';
+import {RefObject, useEffect, useState} from 'react';
+
+// This was created for a special empty case of a component that can have child or
+// be empty, like Collection/Virtualizer/Table/ListView/etc. When these components
+// are empty they can have a message with a tabbable element, which is like them
+// being not empty, when it comes to focus and tab order.
+//
+// This looks at the element's children and determines if any are tabbable.
+export function useHasTabbableChild(ref: RefObject<HTMLElement>): boolean {
+  let [hasTabbableChild, setHasTabbableChild] = useState(false);
+
+  useEffect(() => {
+    if (ref?.current) {
+      // Detect if there are any tabbable elements and update the tabIndex accordingly.
+      let walker = getFocusableTreeWalker(ref.current, {tabbable: true});
+      setHasTabbableChild(!!walker.nextNode());
+    }
+  }, [ref?.current]);
```

Comment:
This causes a lint warning. `React Hook useEffect has a missing dependency: 'ref'. Either include it or remove the dependency array. Mutable values like 'ref.current' aren't valid dependencies because mutating them doesn't re-render the component  react-hooks/exhaustive-deps` I tried fixing it by using just `ref`, but then the code doesn't work. The code this was based on used `useLayoutEffect`, had a MutationObserver, and `ref` worked. https://github.com/adobe/react-spectrum/blob/main/packages/%40react-aria/tabs/src/useTabPanel.ts#L60 Suggestions?

Reply:
correct, this should just be `ref` and you should add isEmpty and isRenderingAnything as I mention in my other comment